### PR TITLE
Add FileTree

### PR DESCRIPTION
**Pull request by stuartmscott**
Extends widget.Tree for displaying a file system hierarchy

### CODE REVIEW

**Review by Jacalz — CHANGES_REQUESTED** [review]
The icon in the README looks broken, but it looks good to me otherwise.

**Review by Jacalz — DISMISSED** [review]
I don't have a clue about what the Sorter() function does, but apart from that it looks very good :)

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Agreed this looks like a great addition, but let's resolve the broken tests before landing

**Comment by AlbinoGeek** [comment]
That's the wrong import path isn't it? (in the example code in the `README`)

**Comment by stuartmscott** [comment]
@AlbinoGeek can you elaborate? That code compiles and runs just fine

**Comment by AlbinoGeek** [comment]
@stuartmscott Based on the [`README`](../blob/master/README.md) I figured the import paths to be used were:

`fyne.io/x/fyne/layout`
`fyne.io/x/fyne/widget`

**Comment by stuartmscott** [comment]
Ah I see what you mean, tho `fyne.io/x/fyne/widget` is a vanity URL for `github.com/fyne-io/fyne-x/widget` so 'wrong' is a little strong. Will fix

**Comment by AlbinoGeek** [comment]
Good point! I apologize for the wording. I figured with the way `go-import-meta` is, the "vanity URL" could do a little more than that (such as versioning, etc.) -- but I see at the moment it is not doing any of these things.

**Comment by stuartmscott** [comment]
No worries :)

This PR is ready for review, tho it can't be merged because the pesky FileIcon is missing from the golden image

File: widget/filetree.go [hunk 1]
```diff
@@ -138,7 +138,6 @@ func NewFileTree(root fyne.URI) *FileTree {
 		} else {
 			// Set file uri to update icon
 			c.Objects[1].(*widget.FileIcon).SetURI(uri)
-			c.Objects[1].(*widget.FileIcon).Resize(c.Objects[1].(*widget.FileIcon).Size())
 			c.Objects[1].(*widget.FileIcon).Refresh()
```
[the remark this answers is not in the document]
Comment (stuartmscott) [reply widget/filetree.go:141]: Ideally Refresh wouldn't be needed at all once SetURI is fixed

[the remark this answers is not in the document]
Comment (stuartmscott) [reply widget/filetree.go]: Done

**Review by andydotxyz — APPROVED** [review]

**Review by andydotxyz — APPROVED** [review]
Sweet!

**Pull request merged**